x/distribution/keeper/validator.go

```diff
@@ -42,30 +42,58 @@ func (k Keeper) incrementValidatorPeriod(ctx sdk.Context, val sdk.Validator) uin
 	}
 
 	// fetch historical rewards for last period
-	historical := k.GetValidatorHistoricalRewards(ctx, val.GetOperator(), rewards.Period-1)
+	historical := k.GetValidatorHistoricalRewards(ctx, val.GetOperator(), rewards.Period-1).PeriodRewards
 
-	// fet new historical rewards
-	k.SetValidatorHistoricalRewards(ctx, val.GetOperator(), rewards.Period, historical.Plus(current))
+	// set new historical rewards with reference count of 1
+	k.SetValidatorHistoricalRewards(ctx, val.GetOperator(), rewards.Period, types.NewValidatorHistoricalRewards(historical.Plus(current), 1))
 
 	// set current rewards, incrementing period by 1
 	k.SetValidatorCurrentRewards(ctx, val.GetOperator(), types.NewValidatorCurrentRewards(sdk.DecCoins{}, rewards.Period+1))
 
 	return rewards.Period
 }
 
+// increment the reference count for a historical rewards vlaue
+func (k Keeper) incrementReferenceCount(ctx sdk.Context, valAddr sdk.ValAddress, period uint64) {
+	historical := k.GetValidatorHistoricalRewards(ctx, valAddr, period)
+	if historical.ReferenceCount > 1 {
+		panic("reference count should never exceed 1")
+	}
+	historical.ReferenceCount++
```

Comment:
because we are `++`ing it here does this mean that `incrementReferenceCount` will panic (on the above condition the second time it is called? maybe elaborate - maybe worth adding in as a comment?